docs/docsite/rst/become.rst

```diff
@@ -202,8 +205,99 @@ or '/bin/chmod' as the allowed commands this will fail with ansible as those
 paths won't match with the temporary file that ansible creates to run the
 module.
 
+.. _become-network:
+
+Become and Networks
+===================
+
+
+network_cli and become
+----------------------
+
+Ansible 2.5 added support for ``become`` to be used to enter `enable` mode (Privileged EXEC mode) on network devices that support it. This replaces the previous ``authorize`` and ``auth_pass`` options in ``provider``.
+
+This functionality requires the host connection type to be using ``connection: network_cli``, in Ansible 2.5 this is limited to ``eos`` and ``ios``.
+
+This allows privileges to be raised for the specific tasks that need them. Adding ``become: yes`` and ``become_method: enable`` informs Ansible to go into privilege mode before executing the task.
+
+If a task fails with the following then it's an indicator that `enable` mode is required:
+
+.. code-block:: console
+
+   Invalid input (privileged mode required)
+
+Which can be enabled for specific tasks as shown (task level):
+
+.. code-block:: yaml
+
+   - name: Gather facts (eos)
+     eos_facts:
+       gather_subset:
+         - "!hardware"
+     become: yes
+     become_method: enable
+
+Or if you wish to be in enable mode for all tasks (play level):
+
+.. code-block:: yaml
+
+   - hosts: eos-switches
+     become: yes
+     become_method: enable
+     tasks:
+       - name: Gather facts (eos)
+         eos_facts:
+           gather_subset:
+             - "!hardware"
+
+FIXME: Can be specified per host
+
+Dropping out of enable mode
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+FIXME: Not possible, as some platforms don't support this
```

Comment:
Maybe not universally, but `become: no` definitely works as expected for platforms whose terminal plugin has an `on_unbecome()` method